packages/common/infra/src/modules/doc/entities/doc.ts

```diff
@@ -1,18 +1,25 @@
 import type { RootBlockModel } from '@blocksuite/blocks';
 
 import { Entity } from '../../../framework';
+import type { WorkspaceService } from '../../workspace';
 import type { DocScope } from '../scopes/doc';
 import type { DocsStore } from '../stores/docs';
 import type { DocMode } from './record';
 
 export class Doc extends Entity {
   constructor(
     public readonly scope: DocScope,
-    private readonly store: DocsStore
+    private readonly store: DocsStore,
+    private readonly workspaceService: WorkspaceService
   ) {
     super();
   }
 
+  /**
+   * for convenience
+   */
+  readonly workspace = this.workspaceService.workspace;
```

Comment:
change to getter? In `WorkspaceService` the workspace is a lazy initialized value.